arch/x86/core/ia32/crt0.S

```diff
@@ -308,10 +308,8 @@ __csSet:
 	outb %al, $0xA1		/* set i8259 slave mask (IRQs 8-15) */
 #endif
 
-	/* Jump to C portion of kernel initialization and never return */
-
-	jmp	z_cstart
-
+	/* call C portion of kernel initialization and never return */
+	call	z_x86_prep_c
```

Comment:
Love it.

Reply:
Eventually, we should nuke z_kernel_arch_init() for all arches